src/nvim/option.c

```diff
@@ -1979,7 +1978,7 @@ void set_option_sctx_idx(int opt_idx, int opt_flags, sctx_T script_ctx)
 }
 
 /// Apply the OptionSet autocommand.
-static void apply_optionset_autocmd(int opt_idx, long opt_flags, long oldval, long oldval_g,
+static void apply_optionset_autocmd(int opt_idx, long opt_flags, OptInt oldval, OptInt oldval_g,
                                     long newval, const char *errmsg)
```

Comment:
newval should also be `OptInt` or any particular reason it is still `long` ?

Reply:
Just to make the minimal/fastest possible PR to unblock @famiu's work. Will address in a follow-up PR.